### DISCUSSION TITLE
Cruise Control Intra broker goals support using the Strimzi Operator

### DISCUSSION THREAD

**Discussion by edubois10** [strimzi/strimzi-kafka-operator#7055] [Q&A] [answered]
Hi Folks,

In the strimzi github page is mentionned that IntraBrokerDiskCapacityGoal, IntraBrokerDiskUsageDistributionGoal are supported goals.
https://github.com/strimzi/strimzi-kafka-operator/blob/main/documentation/modules/cruise-control/con-optimization-goals.adoc. To configure these goals in the Kafka CRD, one need to use the intra.broker.goals configuration as mentionned in https://github.com/linkedin/cruise-control/wiki/Configurations. 

`intra.broker.goals: >
        com.linkedin.kafka.cruisecontrol.analyzer.goals.IntraBrokerDiskCapacityGoal,
        com.linkedin.kafka.cruisecontrol.analyzer.goals.IntraBrokerDiskUsageDistributionGoal`

So far so good. However when I create a KafkaRebalance CRD with one of these goals, I get the following error message:
`message: >-
        Error for request:
        test01-cruise-control.test01.svc:9090/kafkacruisecontrol/rebalance?json=true&dryrun=true&verbose=true&skip_hard_goal_check=true&replica_movement_strategies=PrioritizeSmallReplicaMovementStrategy%2CPostponeUrpReplicaMovementStrategy&concurrent_partition_movements_per_broker=3&concurrent_intra_broker_partition_movements=2&goals=IntraBrokerDiskCapacityGoal%2CPotentialNwOutGoal%2CReplicaDistributionGoal.
        Server returned: Error processing POST request '/rebalance' due to:
        'com.linkedin.kafka.cruisecontrol.exception.KafkaCruiseControlException:
        java.lang.IllegalArgumentException: IntraBrokerDiskCapacityGoal does not
        support balancing action not specifying logdir.'.`

I couldn't find any documentation about "specifying logdir", and also I couldn't find any documentation about IntraBrokerDiskCapacityGoal and IntraBrokerDiskUsageDistributionGoal in the RedHat documentation https://access.redhat.com/documentation/en-us/red_hat_amq/2020.q4/html/using_amq_streams_on_openshift/cruise-control-concepts-str#con-optimization-proposals-str, therefore I wonder if these goals are not supported with strimzi-operator.

**Comment by scholzj** [comment, marked as answer]
Support for Intra-broker disk balancing was added in Strimzi 0.28. The docs show how to use it: https://strimzi.io/docs/operators/latest/full/configuring.html#intra_broker_disk_balancing

**Reply by edubois10** [reply]
Thank you for your quick answer. Unfortunately I was trying on RedHat AMQ Streams version 2.0.1 which is based on strimzi 0.26.x.

**Reply by scholzj** [reply]
AMQ Streams 2.0 is based on Strimzi 0.26 IIRC. @kyguy might know if there is some hack or something how to do it. But I'm not aware of any.

**Reply by kyguy** [reply]
@scholzj is right and unfortunately there are no simple hacks to get this working for older Strimzi/Streams versions. Best to upgrade to Strimzi 0.28 or AMQ Streams 2.1

**Reply by edubois10** [reply]
OK, I see. Thank you @scholzj @scholzj for your help folks.
